src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -997,9 +997,13 @@ OverlayImpl::processValidatorList(
         return true;
     };
 
-    auto key = req.target().substr(prefix.size());
```

Comment:
I'd personally leave this alone.

Reply:
If I leave type-deduction to auto, I'm not 100% sure if it is deduced as `std::string_view`. On line 1006, there is a comparison of `key` with `std::string_view::npos`.

My observation is that construction of `std::string_view` without the second length parameter results in garbage values at the end of `key`. I'm not really sure why that's happening.

Reply:
i missed this comment, thanks for the explanation Scott! yes, that makes sense